lib/archethic/p2p.ex

```diff
@@ -699,12 +754,13 @@ defmodule Archethic.P2P do
 
     # If the results are the same, then we reached consistency
     if length(distinct_elems) == 1 do
-      {:ok, List.first(distinct_elems)}
+      List.first(distinct_elems)
     else
       # If the results differ, we can apply a conflict resolver to merge the result into
       # a consistent response
       resolved_result = conflict_resolver.(distinct_elems)
-      {:ok, resolved_result}
+
+      resolved_result
```

Comment:
I think to reduce boilerplate, you could handle the acceptance criteria there and return {:ok, _} or {:error, _}, then catch on the do_quorum_read to retry or not, like:

```elixir
case  do_quorum(results, conflict_resolver) do
   {:ok, result} ->
     {:ok, result}
   {:error, _} ->
      do_read_quorum(rest, ....)
end
```

Reply:
the problem is we will need to pass nodes to do_quorum and it will be a cyclic dependecy between do_read_quorum ad do_quorum

Reply:
Why will we need to pass nodes to `do_quorum` ?

Reply:
maybe i didn't understand you correctly,  can you explain a bit more your first comment

Reply:
ok I get, it's ok I think